src/phase0/state_transition.rs

```diff
@@ -46,3 +56,134 @@ pub fn apply_block<
 > {
     state
 }
+
+pub fn verify_block_signature<
+    const SLOTS_PER_HISTORICAL_ROOT: usize,
+    const HISTORICAL_ROOTS_LIMIT: usize,
+    const ETH1_DATA_VOTES_BOUND: usize,
+    const VALIDATOR_REGISTRY_LIMIT: usize,
+    const EPOCHS_PER_HISTORICAL_VECTOR: usize,
+    const EPOCHS_PER_SLASHINGS_VECTOR: usize,
+    const MAX_VALIDATORS_PER_COMMITTEE: usize,
+    const PENDING_ATTESTATIONS_BOUND: usize,
+    const MAX_PROPOSER_SLASHINGS: usize,
+    const MAX_ATTESTER_SLASHINGS: usize,
+    const MAX_ATTESTATIONS: usize,
+    const MAX_DEPOSITS: usize,
+    const MAX_VOLUNTARY_EXITS: usize,
+>(
+    state: BeaconState<
+        SLOTS_PER_HISTORICAL_ROOT,
+        HISTORICAL_ROOTS_LIMIT,
+        ETH1_DATA_VOTES_BOUND,
+        VALIDATOR_REGISTRY_LIMIT,
+        EPOCHS_PER_HISTORICAL_VECTOR,
+        EPOCHS_PER_SLASHINGS_VECTOR,
+        MAX_VALIDATORS_PER_COMMITTEE,
+        PENDING_ATTESTATIONS_BOUND,
+    >,
+    signed_block: SignedBeaconBlock<
+        MAX_PROPOSER_SLASHINGS,
+        MAX_VALIDATORS_PER_COMMITTEE,
+        MAX_ATTESTER_SLASHINGS,
+        MAX_ATTESTATIONS,
+        MAX_DEPOSITS,
+        MAX_VOLUNTARY_EXITS,
+    >,
```

Comment:
the type signatures are going to be annoying huh

Reply:
i'd say we try to deal with the long form as see how bad it really does get, but one alternative that may be a bit more workable is to abbreviate:

```rust
pub fn verify_block_signature<
    const A: usize,
    const B: usize,
    const C: usize,
    const D: usize,
    const E: usize,
    const F: usize,
    const G: usize,
    const H: usize,
    const I: usize,
    const J: usize,
    const K: usize,
    const L: usize,
    const M: usize,
>(
    state: BeaconState<
        ...
    >,
    signed_block: SignedBeaconBlock<
        ...
    >,
```


and, we could always consider a macro here...

```rust
with_bounds! {
    pub fn verify_block_signature(state: BeaconState, signed_block: SignedBeaconBlock) {
    }
}
```